### PR TITLE
Faster geth imports, measure times, Ethash log fix

### PR DESCRIPTION
**Pull request by holiman**
* added retries on failing instance-startup,  
* added speedier block import in geth, 
* added testcase time measurements

### CODE REVIEW

File: clients/go-ethereum:master/geth.sh [hunk 1]
```diff
@@ -88,9 +88,12 @@ fi
...
 if [ -d /blocks ]; then
-	for block in `ls /blocks | sort -n`; do
-		/geth $FLAGS import /blocks/$block
-	done
+	cd blocks
+	../geth $FLAGS --nocompaction import `ls | sort -n`
+	cd ..
+#	for block in `ls /blocks | sort -n`; do
+#		/geth $FLAGS import /blocks/$block
+#	done
```
Comment (karalabe) [inline clients/go-ethereum:master/geth.sh]: Any reason for leaving this comment here? Any reason for updating only master?

File: clients/go-ethereum:master/geth.sh [hunk 2]
```diff
@@ -88,9 +88,12 @@ fi
 # Load the remainder of the test chain
 echo "Loading remaining individual blocks..."
 if [ -d /blocks ]; then
-	for block in `ls /blocks | sort -n`; do
-		/geth $FLAGS import /blocks/$block
-	done
+	cd blocks
+	../geth $FLAGS --nocompaction import `ls | sort -n`
+	cd ..
```
Comment (karalabe) [inline clients/go-ethereum:master/geth.sh]: This could be simply

```
(cd blocks && ../geth $FLAGS --nocompaction import `ls | sort -n`)
```

**Comment by karalabe** [comment]
https://github.com/karalabe/hive/blob/master/clients/go-ethereum:stable/geth.sh#L89

https://github.com/karalabe/hive/blob/master/clients/go-ethereum:local/geth.sh#L89

https://github.com/karalabe/hive/pull/39#discussion_r112429761

>  Any reason for updating only master?

**Comment by holiman** [comment]
Kind of, I'm not sure which versions of geth got `--nocompaction` in it.

**Comment by karalabe** [comment]
Well, given that we just released 1.6.0, it's safe to say every one of them has it by now.

**Comment by karalabe** [comment]
Although I see your PR was open way longer. I apologize for not seeing/forgetting about it.

**Comment by holiman** [comment]
Not in 1.5, which is what `stable`  points to : 
https://github.com/ethereum/go-ethereum/blob/release/1.5/cmd/utils/flags.go

**Comment by karalabe** [comment]
Erm, bummer, we need to update that then.

**Comment by holiman** [comment]
Ok, updated

**Pull request merged**